src/NumericalAlgorithms/FiniteDifference/Unlimited.cpp

```diff
@@ -0,0 +1,48 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "NumericalAlgorithms/FiniteDifference/Unlimited.hpp"
+
+#include "NumericalAlgorithms/FiniteDifference/Reconstruct.tpp"
+#include "Utilities/GenerateInstantiations.hpp"
+
+// NOLINTNEXTLINE(modernize-concat-nested-namespaces)
+namespace fd::reconstruction {
+#define DIM(data) BOOST_PP_TUPLE_ELEM(0, data)
+#define DEGREE(data) BOOST_PP_TUPLE_ELEM(1, data)
+
+#define INSTANTIATION(r, data)                                                 \
+  template void reconstruct<UnlimitedReconstructor<DEGREE(data)>>(             \
+      gsl::not_null<std::array<gsl::span<double>, DIM(data)>*>                 \
+          reconstructed_upper_side_of_face_vars,                               \
+      gsl::not_null<std::array<gsl::span<double>, DIM(data)>*>                 \
+          reconstructed_lower_side_of_face_vars,                               \
+      const gsl::span<const double>& volume_vars,                              \
+      const DirectionMap<DIM(data), gsl::span<const double>>& ghost_cell_vars, \
+      const Index<DIM(data)>& volume_extents,                                  \
+      const size_t number_of_variables);
+
+namespace detail {
+GENERATE_INSTANTIATIONS(INSTANTIATION, (1, 2, 3), (2, 4))
```

Comment:
Q) Why `GENERATE_INSTANTIATIONS` was put inside `detail{}` while the definition of `INSTANTIATION` is outside of it?

Reply:
It doesn't really matter. Putting the definition inside the `detail` namespace means I don't need `detail::UnlimitedReconstructor` in the macro definition. But since macros are copy-paste the compiler will see the code inside the `detail` namespace